### PR TITLE
tetragon: Allow persistent enforcement during tetragon restart

### PR DESCRIPTION
**Pull request by olsajiri**
needs https://github.com/cilium/ebpf/pull/1496 (merged, waiting for cilium/ebpf release)

Adding support to configure tetragon to keep existing policy in place - create pins for all maps/programs/links and do not remove them when tetragon exits.

This allows the policy to be in place when tetragon process is gone.

At the moment the configuration is by global flag. I have changes for configuring this through the spec 'options' flag, but first I need first to find a good way how to configure base sensor with same option. I think it's better to have this in place now, the option flag can come later.

### CODE REVIEW

File: cmd/tetragon/main.go [hunk 1]
```diff
@@ -157,6 +157,34 @@ func stopProfile() {
...
+		log.Info("Found bpf leftover instance, removing: %s", old)
+	}
+	if err := os.Rename(path, old); err != nil {
+		return "", err
+	}
+	log.Infof("Found bpf instance: %s, moved to: %s", path, old)
+	return old, nil
+}
+
+func deleteOlBpfDir(path string) {
```
Comment (tpapagian) [inline cmd/tetragon/main.go:177]: Typo: deleteOlBpfDir -> deleteOldBpfDir

File: cmd/tetragon/main.go [hunk 2]
```diff
@@ -157,6 +157,34 @@ func stopProfile() {
...
+	if path == "" {
+		return
+	}
+	if err := os.RemoveAll(path); err != nil {
+		log.Errorf("Failed to remove old bpf isntance '%s': %s\n", path, err)
```
Comment (tpapagian) [inline cmd/tetragon/main.go:182]: Typo: isntance

**Review by tpapagian — APPROVED** [review]
LGTM, thanks!

Just a quick question: Is `link.Pin` also supported in the other types of programs that Tetragon supports (i.e. LSM, Tracing, Uprobes, RawTracepoints)? If yes, I believe that it is a good idea to update those as well. In that case, we also need to have a feature detection function using the same type of program instead of kprobes?

**Comment by olsajiri** [comment]
> LGTM, thanks!
> 
> Just a quick question: Is `link.Pin` also supported in the other types of programs that Tetragon supports (i.e. LSM, Tracing, Uprobes, RawTracepoints)? If yes, I believe that it is a good idea to update those as well. In that case, we also need to have a feature detection function using the same type of program instead of kprobes?

yes, those should be supported as wee, I'll do follow up with that support, thanks

**Review by kkourt — APPROVED** [review]
LGTM, thanks!

**Pull request merged**